src/store/snackbars.tsx

```diff
@@ -0,0 +1,61 @@
+import produce, { Draft } from 'immer'
+import React from 'react'
+import create, { State, StateCreator } from 'zustand'
+
+import { createId } from '@/utils/createId'
+
+type SnackbarIconType = 'success' | 'error' | 'info' | 'warning'
+
+export type DisplaySnackbarArgs = {
+  timeout?: number
+  variant?: 'primary' | 'secondary'
+  iconType?: SnackbarIconType
+  title: string
+  description?: string
+  actionText?: string
+  onActionClick?: () => void
+}
+type Snackbar = {
+  id: string
+} & Omit<DisplaySnackbarArgs, 'time'>
+
+export type SnackbarState = {
+  snackbars: Snackbar[]
+  displaySnackbar: (args: DisplaySnackbarArgs, onActionClick?: () => void) => string
+  updateSnackbar: (id: string, opts: Omit<DisplaySnackbarArgs, 'id'>) => void
+  closeSnackbar: (id: string) => void
+}
+
+const immer = <T extends State>(config: StateCreator<T, (fn: (draft: Draft<T>) => void) => void>): StateCreator<T> => (
```

Comment:
I decided to not use immer after all.  I don't want to overcomplicate things for now.  In the future probably we could create `middleware.ts` and move this function there and use it in the root store.